tests/test_adjoint.py

```diff
@@ -110,7 +110,7 @@ def test_adjoint_J(self, shape, space_order):
         term2 = linalg.norm(du.data.reshape(-1))**2
         info('<Jx,y>: %f, <x, J^Ty>: %f, difference: %4.4e, ratio: %f'
              % (term1, term2, (term1 - term2)/term1, term1 / term2))
-        assert np.isclose((term1 - term2)/term1, 0., rtol=1.e-10)
+        assert np.isclose((term1 - term2)/term1, 0., rtol=1.e-10, atol=0.)
```

Comment:
Ok so I re-read this with frech brain.

`absolute(a - b) <= (atol + rtol * absolute(b))`

`b` is always `0` so `rtlo` is uselss (0*rtol always 0), but `atol=0` is completley unfeasible never gonna be that exact. test in double precision so should be 

`np.isclose((term1 - term2)/term1, 0., atol=1e-12)`

which test 

`abs(a-b) = abs(a) <= 1e-12)` or whatever `atol` you think is consistent, current one is default which is 1e-8

Reply:
> Ok so I re-read this with frech brain.

I read this as "French brain" :laughing:

The biggest problem I see in doing it this way is if term1 is really close to 0, it is scary to divide by term1.

Reply:
term1 should never be close to zero unless setup weirdly